### ISSUE TITLE
[16.0][POS] Discount enabled without Price Control rights

### ISSUE THREAD

**Issue by flachica** [odoo/odoo#120030]
Impacted versions: 16.0

Steps to reproduce: In Settings enable: `Multi Employees per Session`, `Line Discounts` and create one user with administrator role and other with only user. Select user role

Current behavior: `% Disc.` Numpad button is enabled. With the user role in the POS, just as it is not allowed to change the price, it should not be allowed to add discounts, as the effect is similar.

Expected behavior: `% Disc.` Numpad button is disabled as `Price` button

**Comment by pedrambiria** [comment]
Thank you for bringing this to our attention.

You're absolutely right that from a control perspective, it might seem intuitive to restrict the discount functionality in a similar way to the price change. However, the "Line Discounts" feature is designed to allow any user, regardless of their role, to apply discounts on an order. This is part of its intended functionality and is working as expected.

I understand that this might be an issue if you don't want certain users to have the ability to apply discounts. In that case, it's recommended to establish clear policies and guidelines for your team. Shop managers should clearly communicate what's expected from their team members and if necessary, disable certain features for specific roles or users.

Importantly, any discount applied is recorded on the POS orders, along with the name of the cashier who processed the transaction. This means any misuse of the discount functionality will be clearly visible to shop managers, who can then take appropriate action.

I hope this clarifies the situation. Please feel free to share more details if you have further concerns or suggestions. We always appreciate feedback from our users to improve our product.

Best,

**Comment by flachica** [comment]
Of course, that's precisely what I expected Odoo to do. But it turns out that it is impossible "disable certain features for specific roles or users". There is no group to prevent these operations.

**Comment by pedrambiria** [comment]
Indeed, you've highlighted an important point. Currently, Odoo's Point of Sale (POS) does not support group-based restrictions for certain operations on the front end, including price control. 

Adding such restrictions at this level would change the expected behavior of the system, which could potentially cause issues for other users who rely on the current functionality. This is a significant consideration in a widely used system like Odoo.

However, Odoo is an open-source platform and one of its main advantages is its flexibility and openness to customization. If you have specific business needs that aren't met by the standard features, you can customize Odoo to suit those needs.

In this case, if you need to restrict certain operations based on user roles, you can potentially achieve this by developing a custom module or extending an existing one. If you need assistance with customization, consider reaching out to experienced Odoo developers or consulting services.

I hope this clarifies the situation. Please feel free to provide more details if you have further questions or concerns.
